bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/controller/AIChatController.java

```diff
@@ -0,0 +1,120 @@
+
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+import jakarta.annotation.Resource;
+import java.util.List;
+
+@Tag(name = "AI Chat Controller")
+@RestController
+@RequestMapping("/ai/chat/")
+public class AIChatController {
+
+    @Resource
+    private AIChatService chatService;
+
+    @Operation(summary = "platforms", description = "Get all platforms")
+    @GetMapping("/platforms")
+    public ResponseEntity<List<PlatformVO>> platforms() {
+        return ResponseEntity.success(chatService.platforms());
+    }
+
+    @Operation(summary = "platforms", description = "Get authorized platforms")
+    @GetMapping("/platforms/authorized")
+    public ResponseEntity<List<PlatformAuthorizedVO>> authorizedPlatforms() {
+        return ResponseEntity.success(chatService.authorizedPlatforms());
+    }
+
+    @Operation(summary = "platforms", description = "Add authorized platforms")
+    @PutMapping("/platforms")
+    public ResponseEntity<PlatformVO> addAuthorizedPlatform(@RequestBody PlatformReq platformReq) {
+        PlatformDTO platformDTO = PlatformConverter.INSTANCE.fromReq2DTO(platformReq);
+        return ResponseEntity.success(chatService.addAuthorizedPlatform(platformDTO));
+    }
+
+    @Operation(summary = "platforms", description = "Delete authorized platforms")
+    @DeleteMapping("/platforms/{platformId}")
+    public ResponseEntity<Integer> deleteAuthorizedPlatform(@PathVariable Long platformId) {
+        int code = chatService.deleteAuthorizedPlatform(platformId);
+        if (code != 0) {
+            return ResponseEntity.error(ResponseStatus.PARAMETER_ERROR, "Permission denied");
+        }
+        return ResponseEntity.success(0);
+    }
+
+    @Operation(summary = "new threads", description = "Create a chat threads")
+    @PutMapping("/platforms/{platformId}/threads")
+    public ResponseEntity<ChatThreadVO> createChatThreads(@PathVariable Long platformId, @RequestParam String model) {
+        return ResponseEntity.success(chatService.createChatThreads(platformId, model));
+    }
+
+    @Operation(summary = "delete threads", description = "Delete a chat threads")
+    @DeleteMapping("platforms/{platformId}/threads/{threadId}")
+    public ResponseEntity<Integer> deleteChatThreads(@PathVariable Long platformId, @PathVariable Long threadId) {
```

Comment:
I’m not quite sure what the most appropriate return format here is—an empty `success()`, a `code`, or a `boolean`.

Reply:
`ResponseEntity<Boolean>` is fine